### PR TITLE
Make gossip use channel names on the wire, not channel name hashes

### PR DESCRIPTION
**Pull request by dpw**
Addresses #745.

### CODE REVIEW

**Comment by rade** [comment]
protocol version needs bumping.

File: router/gossip.go [hunk 1]
```diff
@@ -85,23 +85,23 @@ type GossipChannel struct {
 	ourself      *LocalPeer
 	routes       *Routes
 	name         string
-	hash         uint32
 	gossiper     Gossiper
 	senders      connectionSenders
 	broadcasters peerSenders
 }
 
+// Type of GossipChannels in the router
```
Comment (rade) [inline router/gossip.go]: that comment seems superfluous

File: router/gossip.go [hunk 2]
```diff
@@ -123,13 +123,13 @@ func (router *Router) SendAllGossipDown(conn Connection) {
...
-	if err := decoder.Decode(&channelHash); err != nil {
+	var channelName string
+	if err := decoder.Decode(&channelName); err != nil {
 		return err
 	}
-	channel, found := router.GossipChannels[channelHash]
+	channel, found := router.GossipChannels[channelName]
 	if !found {
-		return fmt.Errorf("[gossip] received unknown channel with hash %v", channelHash)
+		return fmt.Errorf("[gossip] received unknown channel with name %v", channelName)
```
Comment (rade) [inline router/gossip.go]: `'%s'` would be better, I think.

**Comment by dpw** [comment]
Re-spun with comments addressed.

**Pull request merged**